tensorflow/compiler/xla/service/pattern_matcher.h

```diff
@@ -171,6 +176,27 @@ bool Match(Value* value, const Pattern& pattern,
   return pattern.Match(value, option);
 }
 
+// Recursively requires all operands of the top-level operation prescribed in
+// pattern (but not the top-level operation itself) to have one user. The
+// behavior is identical to calling Match(value, pattern) with WithOneUser()
+// applied to all prescribed operands at all levels in pattern.
+//
+// Example:
+// p0 = parameter(0)
+// add = add(p0, p0)
+// mul = multiply(p0, p0)
+//
+// MatchSingleUserOperands(p0, m::Op())) -> true.
+// MatchSingleUserOperands(add, m::Add()) -> true.
+// MatchSingleUserOperands(add, m::Add(m::Op(), m::Op())) -> false.
+
+template <typename Value, typename Pattern>
+bool MatchSingleUserOperands(Value* value, const Pattern& pattern) {
```

Comment:
Maybe `MatchSingleUseOnly` is more clear that it complements `Match`? Also no whitespace between the docstring and the function, and maybe add an example with `Match` vs. `MatchSingleUse`.